src/azure-cli/azure/cli/command_modules/acr/connected_registry.py

```diff
@@ -0,0 +1,494 @@
+                                  registry_name,
+                                  yes=False,
+                                  resource_group_name=None):
+
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+
+    user_confirmation("Are you sure you want to delete the Connected Registry '{}' from '{}'?".format(
+        connected_registry_name, registry_name), yes)
+    try:
+        result = client.delete(resource_group_name, registry_name, connected_registry_name)
+        msg = "Connected registry successfully deleted. Please cleanup your sync tokens and scope maps. " + \
+              "Run the following commands for cleanup: \n\t" + \
+              "az acr token delete -n <sync token> -r {}\n\t".format(registry_name) + \
+              "az acr scope-map delete -n <sync scope map> -r {}\n".format(registry_name) + \
+              "Run the following command on all ascendency to remove the deleted registry gateway access: \n\t" + \
+              "az acr scope-map update -n <parent sync scope map> -r {} --remove-gateway {}".format(
+                  registry_name, " ".join([connected_registry_name] + DEFAULT_GATEWAY_SCOPE))
+        logger.warning(msg)
+        return result
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def acr_connected_registry_deactivate(cmd,
+                                      client,
+                                      connected_registry_name,
+                                      registry_name,
+                                      yes=False,
+                                      resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+
+    user_confirmation("Are you sure you want to deactivate the Connected Registry '{}' from '{}'?".format(
+        connected_registry_name, registry_name), yes)
+    return client.deactivate(subscription_id=subscription_id,
+                             resource_group_name=resource_group_name,
+                             registry_name=registry_name,
+                             connected_registry_name=connected_registry_name)
+
+
+def acr_connected_registry_list(cmd,
+                                client,
+                                registry_name,
+                                parent=None,
+                                no_children=False,
+                                resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+    connected_registry_list = list(client.list(resource_group_name, registry_name))
+    result = []
+    if no_children:
+        if parent:
+            result = [registry for registry in connected_registry_list \
+                if registry.parent.id is not None and registry.parent.id.endswith(parent)]
```

Comment:
>endswith(parent) [](start = 73, length = 16)

can it handle the following cases?

parent
myparent

endswith("/"+parent)

Reply:
No, but is that possible? this is reading from RP response. It's not checking users input.

Reply:
I could add cover that scenario if it's possible.

Reply:
I mean if there are two connected registry resources.

connectedRegistries/parent
connectedRegistries/myparent

customer run `az acr connected-registry list --parent parent`, the above code  will return the result from `connectedRegistries/myparent` as well.



---
In reply to: [545675378](https://github.com/Azure/azure-cli/pull/16238#discussion_r545675378) [](ancestors = 545675378)